pulsar-jms/src/main/java/com/datastax/oss/pulsar/jms/Utils.java

```diff
@@ -457,4 +464,53 @@ static int mapPriorityToPartitionNonLinearly(int priority, int numPartitions) {
     }
     return result;
   }
+
+  /**
+   * Utility to convert path configurations into correct formats for the Apache Pulsar client.
+   * Only values for configuration keys that end with "path", case-insensitive, will be modified.
+   * If the value is a String and begins with "base64:", case-sensitive, it will be base64 decoded then written to a
+   * temp file. The temp file permissions will only allow the current user to access the file. The temp file will be
+   * deleted on JVM exit.
+   * If the value is a String and begins with "classpath:", case-sensitive, the complete filepath will be resolved.
+   * @param configuration - the configuration map to modify in place
+   * @return list of paths representing files created by this method
+   */
+  static List<Path> writeEncodedPathConfigsToTempFiles(Map<String, Object> configuration) {
+    List<Path> createdFiles = new ArrayList<>();
+    for (Map.Entry<String, Object> entry : configuration.entrySet()) {
+      String key = entry.getKey();
+      Object value = entry.getValue();
+      if (key.toLowerCase().endsWith("path") && value instanceof String) {
+        String originalPath = (String) value;
+        String finalPath;
+        if (originalPath.startsWith("base64:")) {
+          try {
+            Path file = Files.createTempFile("pulsar-jms.", ".tmp");
+            file.toFile().deleteOnExit();
+            Files.setPosixFilePermissions(file, PosixFilePermissions.fromString("rw-------"));
+            createdFiles.add(file);
+            String encoded = originalPath.replace("\n", "").replace("\r", "").trim();
+            encoded = encoded.substring("base64:".length());
+            Files.write(file, Base64.getDecoder().decode(encoded));
+            finalPath = file.toAbsolutePath().toString();
+            log.info("Decoded {} to temporary file {}", key, finalPath);
+          } catch (IOException ex) {
+            throw new RuntimeException(
+                    "Cannot decode base64 " + key + " and create temporary file: " + ex, ex);
+          }
+        } else if (originalPath.startsWith("classpath:")) {
+          URL path = Utils.class.getClassLoader().getResource(originalPath.substring("classpath:".length()));
+          if (path == null) {
+            throw new IllegalArgumentException("File " + originalPath + " not found for " + key);
+          }
+          finalPath = path.getPath();
+          log.info("Resolved {} to {}", key, finalPath);
```

Comment:
This wouldn't work for all cases. For example, for a resource inside a jar file. For a generic solution, it would be necessary to copy to a temp file.

Reply:
I agree, we must copy to a temporary file, because with this way it works only if you have a directory (like an exploded jar) in the classpath.

I believe that it is always better to download the resource and put it into a temporary file

Reply:
Thanks for your feedback. I didn't realize that was necessary. I'll fix it now.